projects/plugins/jetpack/_inc/lib/class.media-extractor.php

```diff
@@ -450,14 +450,19 @@ protected static function reduce_extracted_images( $images ) {
 			if ( empty( $image['src'] ) ) {
 				continue;
 			}
+			$ret_image = array(
+				'url' => $image['src'],
+			);
+			if ( ! empty( $image['src_height'] ) || ! empty( $image['src_width'] ) ) {
+				$ret_image['src_width']  = $image['src_width'] ?? '';
+				$ret_image['src_height'] = $image['src_height'] ?? '';
+			}
 			if ( ! empty( $image['alt_text'] ) ) {
-				$ret_images[] = array(
-					'url'      => $image['src'],
-					'alt_text' => $image['alt_text'],
-				);
+				$ret_image['alt_text'] = $image['alt_text'];
 			} else {
-				$ret_images[] = $image['src'];
+				$ret_image = $image['src'];
```

Comment:
I think this could overwrite `src_width` and `src_height`, is this intentional?

Reply:
Great question! yes, because the way that `$extract_alt_text` was implemented, when it is set to true, then the results are returned as an array of associative arrays. When it is set to false, then a simple array of urls is returned. I am fairly certain that only ES indexing is using the `$extract_alt_text` option, so I decided to just expand on that, in order to keep other uses backwards compatible.